### PR TITLE
[RISCV] Refactor GPRF64 register class to make it usable for Zacas.

### PR DESCRIPTION
**Pull request by topperc**
-Rename to GPRPair.
-Rename registers to be named like X10_X11 instead of X10_PD. Except X0
 which is now X0_Pair since it is not paired with X1.
-Use unknown size and offset for the subreg indices. This might
 be a functional change, but does not affect any lit tests.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVRegisterInfo.td [hunk 1]
```diff
@@ -546,33 +549,36 @@ def DUMMY_REG_PAIR_WITH_X0 : RISCVReg<0, "0">;
...
-def GPRPF64 : RegisterClass<"RISCV", [f64], 64, (add
-    X10_PD, X12_PD, X14_PD, X16_PD,
-    X6_PD,
-    X28_PD, X30_PD,
-    X8_PD,
-    X18_PD, X20_PD, X22_PD, X24_PD, X26_PD,
-    X0_PD, X2_PD, X4_PD
+let RegInfos = RegInfoByHwMode<[RV64], [RegInfo<64, 64, 64>]>,
+    DecoderMethod = "DecodeGPRPairRegisterClass" in
+def GPRPair : RegisterClass<"RISCV", [f64], 64, (add
```
Comment (dtcxzyw) [inline llvm/lib/Target/RISCV/RISCVRegisterInfo.td:575]: Emm, looks like it cannot be used by zacas in rv32.
```suggestion
def GPRPair : RegisterClass<"RISCV", [untyped], 64, (add
```
Is it OK to use `f64` as the `regTypes`? It is wired for me.

Comment (topperc) [reply llvm/lib/Target/RISCV/RISCVRegisterInfo.td:575]: Yeah that needs to be fixed. I didn't do CodeGen for Zacas yet so it didn't matter.

**Review by asb — APPROVED** [review]
This LGTM. It's a good refactoring, and I think the in-tree testing for Dinx is better than it was.

**Review by dtcxzyw — APPROVED** [review]
LGTM.

**Pull request merged**